Integrations/AlienVault_OTX_v2/AlienVault_OTX_v2.py

```diff
@@ -0,0 +1,875 @@
+    raw, ec = alienvault_search_pulses()
+
+    human_readable = ''
+    # Add three tables at most by threshold 3, change value for more
+    counter = 1
+    thresh = 3
+    page_num = demisto.args().get('page')
+    pulses = ec['AlienVaultOTX'].get('Pulses')
+    total_num_pulse = len(pulses)
+    for pulse in pulses:
+        human_readable += tableToMarkdown(t=pulse,
+                                          name=f'Pulse number {counter}/{total_num_pulse} from page {page_num}')
+        counter += 1
+        if counter > thresh:
+            break
+
+    return_outputs(readable_output=human_readable,
+                   outputs=ec,
+                   raw_response=raw)
+
+
+# alienvault-get-pulse-details command
+def alienvault_get_pulse_details():
+    api_path = {
+        'source': 'pulses',
+        'value': demisto.args().get('pulse-id'),
+    }
+
+    # raw_data
+    raw = http_request(source=api_path['source'],
+                       value=api_path['value'])
+
+    ec = {
+        'AlienVaultOTX': {
+            'Pulses': {
+                'Description': raw.get('description'),
+                'Created': raw.get('created'),
+                'Author': {
+                    'Username': raw.get('author').get('username')
+                },
+                'ID': raw.get('id'),
+                'Name': raw.get('name'),
+                'Tags': raw.get('tags'),
+                'TargetedCountries': raw.get('targeted_countries')
+            }
+        }
+    }
+
+    ec = remove_none(ec)
+
+    return raw, ec
+
+
+def alienvault_get_pulse_details_command():
+    raw, ec = alienvault_get_pulse_details()
+
+    human_readable = ''
+    if ec.get('AlienVaultOTX'):
+        if ec.get('AlienVaultOTX').get('Pulses'):
+            if ec.get('AlienVaultOTX').get('Pulses').get('Author'):
+                # Table 1
+                human_readable = tableToMarkdown(t=ec['AlienVaultOTX']['Pulses']['Author'],
+                                                 name='Pulse author')
+            if ec.get('AlienVaultOTX').get('Pulses').get('Tags'):
+                # Table 2
+                human_readable = tableToMarkdown(t={'Tags': ec['AlienVaultOTX']['Pulses']['Tags']},
+                                                 name='Tags')
+            key_not_need = ['Author', 'Tags']
+            ec_other = {k: ec['AlienVaultOTX']['Pulses'][k] for k in ec['AlienVaultOTX']['Pulses'] if k not in key_not_need}
+            # Table 3
+            human_readable += tableToMarkdown(t=ec_other,
+                                              name='Pulses General')
+
+    return_outputs(readable_output=human_readable,
+                   outputs=ec,
+                   raw_response=raw)
+
+
+# url command
+def url():
+    api_path = {
+        'source': 'indicators',
+        'command': 'url',
+        'value': demisto.args().get('url'),
+        'section': 'general'
+    }
+
+    # raw_data
+    raw = http_request(source=api_path['source'],
```

Comment:
creating a dictionary and passing values one by one is wasteful
```suggestion
    raw = http_request(**api_path)
```

Reply:
did it for all the placed I called the function.